### DISCUSSION TITLE
How I can disable caching sql queries?

### DISCUSSION THREAD

**Discussion by sheldygg** [sqlalchemy/sqlalchemy#11627] [Usage Questions]
I make migration via alembic, then restart my application and start receive this error

(sqlalchemy.dialects.postgresql.asyncpg.InvalidCachedStatementError) <class 'asyncpg.exceptions.InvalidCachedStatementError'>: cached statement plan is invalid due to a database schema or configuration change (SQLAlchemy asyncpg dialect will now invalidate all prepared caches in response to this exception)
[SQL: SELECT ...

How i can fix this?

**Comment by CaselIT** [comment]
Hi,

If you restart your application after the migration has taken place that error should not happen.
In any case it's documented in the asyncpg docs https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#prepared-statement-cache

**Reply by sheldygg** [reply]
As far I see, sqlalchemy has its own cache implementation, since the source code is quite difficult to read, I couldn't figure it out to the fullest.
I use sqlalchemy with asyncpg.
For disabling cache in sqlalchemy I must pass `prepared_statement_cache_size` to url, I see what this parameter not disable cache in asyncpg, and `asyncpg.connection.Connection.statement_cache_size == 100`

For turning off asyncpg cache I must pass `connect_args={"statement_cache_size": 0}` to `create_async_engine`
This is bug?

**Reply by sheldygg** [reply]
And how I can pass `prepared_statement_cache_size` to `AsyncAdapt_asyncpg_connection` without URL?

**Reply by CaselIT** [reply]
It's documented here https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#prepared-statement-cache

**Comment by PiyushPratap10** [comment]
Hey, I recently encountered this Exception when I ran a database migration where I changed the datatype of a field from enum to str. I tried the solutions provided by Deepseek and documentation also didn't mentioned specifically how to deal with this situation. So, I used a brute force approach of  - "Uninstalling sqlmodel and sqlalchemy and then reinstalling them and re running the server and it worked! "

**Reply by CaselIT** [reply]
I don't know about sqlmodel, but sqlalchemy doesn't persist caches, so likely what fixed it was the restart of the server

**Comment by prostopank** [comment]
Hi! If you're using PgBouncer, restarting it resolved the issue for me. Here's a code snippet you can wrap in a function and run after applying migrations to restart PgBouncer and clear its cache.

        conn = psycopg2.connect(
            dbname="pgbouncer",
            user="user",
            password="password",
            host="localhost",
            port=6432,
        )
        conn.autocommit = True

        with conn.cursor() as cursor:
            cursor.execute("RELOAD;")

        conn.close()
